File: frontend/src/types/task.ts
export interface DailyTask {
  id: string;
  user_id: string;
  week_id: string | null;
  task_text: string;
  category: string | null;
  time_estimate: number;
  task_date: string;
  is_done: boolean;
  completed_at: string | null;
  source: string;
  sort_order: number;
  created_at: string;
}

export interface DailyTaskCreate {
  task_text: string;
  category?: string;
  time_estimate?: number;
  assigned_date?: string;
}

export interface DailyTaskUpdate {
  task_text?: string;
  category?: string;
  time_estimate?: number;
  sort_order?: number;
}

export interface CurrentWeekInfo {
  week_number: number;
  title: string;
  theme: string;
}

export interface TodayTasksResponse {
  tasks: DailyTask[];
  completion_pct: number;
  done_count: number;
  total_count: number;
  message?: string | null;
  current_week?: CurrentWeekInfo | null;
}

export interface DayGrid {
  date: string;
  day_name: string;
  is_today: boolean;
  tasks: DailyTask[];
}

export interface WeeklyGridResponse {
  week_start: string;
  days: DayGrid[];
}

File: backend/app/schemas/task.py
from __future__ import annotations

from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, Field


class DailyTaskCreate(BaseModel):
    task_text: str
    category: Optional[str] = Field(None, max_length=100)
    time_estimate: Optional[int] = None
    assigned_date: date


class DailyTaskUpdate(BaseModel):
    task_text: Optional[str] = None
    category: Optional[str] = Field(None, max_length=100)
    time_estimate: Optional[int] = None
    sort_order: Optional[int] = None


class DailyTaskResponse(BaseModel):
    id: str
    user_id: str
    week_id: Optional[str] = None
    task_text: str
    category: Optional[str] = None
    time_estimate: int
    task_date: date
    is_done: bool
    completed_at: Optional[datetime] = None
    source: str
    sort_order: int
    created_at: datetime

    model_config = {"from_attributes": True}
